Fix tf.image.central_crop return zero dimension

yongtang opened the pull request:
This fix tries to address the issue raised in #10315 where tf.image.central_crop will return (0, 0, 3) when input is (240, 320, 3).

The issue is that the calculation was done by dividing
```
fraction_offset = int(1 / ((1 - central_fraction) / 2.0))
```
which causes deviation when accuracy is not enough.

This fix did the calculation in double (and use multiply) until the last step.

A test case has been added.

This fix fixes #10315.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Fix tf.image.central_crop return zero dimension

This fix tries to address the issue raised in 10315
where tf.image.central_crop will return (0, 0, 3)
when input is (240, 320, 3).

The issue is that the calculation was done by dividing
```
fraction_offset = int(1 / ((1 - central_fraction) / 2.0))
```
which causes deviation when accuracy is not enough.

This fix cast the calculation to double until the last step.

This fix fixes 10315.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Add test case for tf.image.central_crop return zero dimension

This commit adds test case for 10315.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

tensorflow-jenkins commented:
Can one of the admins verify this patch?

jhseu commented:
Jenkins, test this please

jhseu reviewed (approved):
(Testing to see if this fixes the Jenkins issue)

jhseu commented:
Jenkins, test this please

jhseu commented:
Jenkins, test this please

martinwicke commented:
Rerun of android http://ci.tensorflow.org/job/tensorflow-pull-requests-android/6612

yifeif commented:
android rerun again http://ci.tensorflow.org/job/tensorflow-pull-requests-android/6673/console

drpngx commented:
Trying again. Jenkins, test this please.

yongtang commented:
Thanks @drpngx for the help. The Jenkins test failure was caused by `//tensorflow/tools/api/tests:api_compatibility_test` as the `--update_goldens True` was not run. 

I have rebased and updated the PR with goldens updated. Please take a look and sorry for the inconvenience.

drpngx commented:
Thanks! (The api compatibility test was my mistake)

Jenkins, test this please.

drpngx commented:
Still a probelm with the golden, can you take a look?

yongtang commented:
Thanks @drpngx. The api compatibility error was caused by the recent updates from other PRs.

I have rebased and updated the PR. Now the golden error should be gone. Please take a look.

drpngx commented:
Jenkins, test this please.

yongtang commented:
The Jenkins error is `//tensorflow/python/kernel_tests:fft_ops_test`. Not sure about it though I tend to think it is unrelated.

drpngx commented:
Yes, we have the fix internally. Pushing that.
